### ISSUE TITLE
The `make type-check` target does not run `stack` (only `cabal` is supported)

### ISSUE THREAD

**Issue by wenkokke** [agda/agda#4883]
The `make type-check` target still runs a bare `cabal v1` command. Unfortunately, this wasn't addressed by the recent #4841.

**Comment by rwe** [comment]
Yes, unfortunately that PR only fixed up the existing GHC etc. spawns (and it even missed some that are spawned indirectly through the test suites). There had not been a stack variation implemented for that goal at all and I didn't add new ones. But this is a good issue to keep open. There's a lot of inconsistency in those makefiles; unfortunately make is a _terrible_ tool for complex build configurations, despite its ubiquity.

**Comment by andreasabel** [comment]
> unfortunately make is a _terrible_ tool

Good enough to have won the 2002 ACM software system award. :-D 

A discussion on just type-checking with stack is here: 
- [ ] https://github.com/commercialhaskell/stack/issues/977

**Comment by andreasabel** [comment]
There, the suggested workaround for stack to get just type-checking is: `stack ghci "--ghci-options=-e 1"`.

**Comment by andreasabel** [comment]
Looks like I finally know how to make progress on this issue: https://github.com/haskell/cabal/issues/7368#issuecomment-1025176917

EDIT: I can make progress on the "cabal v1 command" by changing it to a "cabal v2 command".  But this issue seems about a `type-check` goal for `stack`.

**Comment by andreasabel** [comment]
> There, the suggested workaround for stack to get just type-checking is: `stack ghci "--ghci-options=-e 1"`.

I tried
```
$ stack ghci --main-is Agda:exe:agda --ghci-options='-e 1'
```
but it does not compare in speed to the cabal goal `make type-check`.
Neither does:
```
$ stack build --fast
```

So I come to the conclusion that for now, the goal is not attainable.  [Why not develop with `cabal`?]
`stack` enthusiasts, if you find a solution, please contribute it!

**Comment by asr** [comment]
With `stack` you should use something like (`cabal` and `stack` had similar compilation times on my setup):

```
stack build --ghc-options -fno-code --ghc-options -fwrite-interface
```

**Comment by andreasabel** [comment]
Ah indeed, after `stack build --dependencies-only` your command "succeeds" (in the sense that it type-checks all modules) and it is as fast the `cabal` version of it.

@asr: Do you want to make this happen?  
One should make sure, like in the `cabal` case, that this type-check-only build does not overwrite the interface files of a regular build.

**Comment by asr** [comment]
> One should make sure, like in the `cabal` case, that this type-check-only build does not overwrite the interface files of a regular build.

I don't know how to change `stack` build directory. The current approach of coping `stack` default directory to the desired directory (e.g. `dist-2.6.5-ghc-9.8.1--no-code`) doesn't work because it  overwrite the interface files of a regular build when using `stack`.

**Comment by asr** [comment]
It seems we could use the option `--work-dir`. I'll make a try.

**Comment by andreasabel** [comment]
@asr: Did you get a chance to work on this?

**Comment by asr** [comment]
No. I switched to `cabal` because its recent versions fixed some issues.

**Comment by andreasabel** [comment]
I find with the advent of haskell-language-server in VSCode the `make type-check` became obsolete.
I am fixing all the type errors in VSCode.
One can also get fast type-checking with `cabal repl` or `stack repl`.
